src/crypto/mod.rs

```diff
@@ -45,6 +48,66 @@ pub struct SigData {
     pub q2: [u8; 384],
 }
 
+/// EC KT-I Public Key, the x-coordinate followed by
+/// the y-coordinate (on the RFC 6090P-256 curve),
+/// 2 x 32 bytes.
+/// A.4, Table 7
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct EcdsaPubKey {
+    pub x: [u8; 32],
+    pub y: [u8; 32],
+}
+
+impl EcdsaPubKey {
+    /// Cast an instance from a byte slice.
+    pub fn cast(raw: &[u8; size_of::<Self>()]) -> &Self {
+        unsafe { transmute(raw) }
```

Comment:
Do we really need this transmute? Is this a performance bottleneck? All the fancy ways of turning one fixed-sized array into are still unstable (https://github.com/rust-lang/rust/issues/90091), but we can do this instead:
```rust
        let (x, y) = z.split_at(32);
        Foo {
            x: x.try_into().unwrap(),
            y: y.try_into().unwrap(),
        }
    }
```
At least in self-contained examples this still appears to generate the ideal assembly, with no branches or panic handling.

Reply:
Sure I can do that thanks